platform/core/src/services/HangingProtocolService/HangingProtocolService.ts

```diff
@@ -1,24 +1,35 @@
-import uuidv4 from '../../utils/uuidv4';
-import pubSubServiceInterface from '../_shared/pubSubServiceInterface';
+import { PubSubService } from '../_shared/pubSubServiceInterface';
 import sortBy from '../../utils/sortBy';
 import ProtocolEngine from './ProtocolEngine';
 import StudyMetadata from '../../types/StudyMetadata';
 import IDisplaySet from '../DisplaySetService/IDisplaySet';
-import { HangingProtocol } from '../../types';
-
-const EVENTS = {
-  STAGE_CHANGE: 'event::hanging_protocol_stage_change',
-  PROTOCOL_CHANGED: 'event::hanging_protocol_changed',
-  NEW_LAYOUT: 'event::hanging_protocol_new_layout',
-  CUSTOM_IMAGE_LOAD_PERFORMED:
-    'event::hanging_protocol_custom_image_load_performed',
-  HANGING_PROTOCOL_APPLIED_FOR_VIEWPORT:
-    'event::hanging_protocol_applied_for_viewport',
-};
+import { HangingProtocol, CommandsManager, Services } from '../../types';
 
 type Protocol = HangingProtocol.Protocol | HangingProtocol.ProtocolGenerator;
 
-class HangingProtocolService {
+const DEFAULT_VIEWPORT_OPTIONS: HangingProtocol.ViewportOptions = {
+  toolGroupId: 'default',
+  viewportType: 'stack',
+};
+
+export default class HangingProtocolService extends PubSubService {
+  static EVENTS = {
+    // The PROTOCOL_CHANGED event is fired when the protocol changes
+    // and should be immediately applied
+    PROTOCOL_CHANGED: 'event::hanging_protocol_changed',
+    // The RESTORE_PROTOCOL event is fired instead of a changed event to indicate
+    // that an earlier state has been restored as part of a state update, but
+    // is not being directly re-applied, but just restored.
+    RESTORE_PROTOCOL: 'event::hanging_protocol_restore',
+    // The layout has been decided for the hanging protocol - deprecated
+    NEW_LAYOUT: 'event::hanging_protocol_new_layout',
+    // Equivalent to PROTOCOL_CHANGED, except for stage instead.  Moslty
+    // this can be ignored in favour of protocol changed.
+    STAGE_ACTIVATION: 'event::hanging_protocol_stage_activation',
```

Comment:
STAGE_CHANGED?

Reply:
No, it is stage activation, updating the docs.